### ISSUE TITLE
Workflows with `stratagy.max-parallel` that have a variable causes an issue

### ISSUE THREAD

**Issue by BlackDex** [woodruffw/github-actions-models#35]
If you have a workflow which has a variable as value for the `max-parallel` option, it causes an issue with zizmor and stops working.

When I set the value to an actual int it doesn't complain.

This works just fine on GitHub actions as far as i know.
Though I know `act` also as an issue with this.

And there are plenty of examples on GitHub which use this: https://github.com/search?q=path%3A%2F%5C.github%5C%2Fworkflows%5C%2F.*%2F+AND+%2Fmax-parallel.*%5C%24%5C%7B%5C%7B%2F&type=code

**workflow:**
```yaml
name: Test Workflow
permissions: {}

on: ['push']

# LIMIT = true|false

jobs:
  test:
    name: Test job
    runs-on: ubuntu-24.04
    strategy:
        # max-parallel: 1
        max-parallel: ${{ vars.LIMIT && 1 || 8 }}
    steps:
      - name: Test step
        shell: bash
        run: |
          echo "Hello World!"
```

**output:**
```bash
failed to register input: .github/workflows/test.yml

Caused by:
    0: invalid GitHub Actions workflow: file://.github/workflows/test.yml
       
       Caused by:
           jobs: data did not match any variant of untagged enum Job at line 9 column 3
    1: invalid GitHub Actions definition: file://.github/workflows/test.yml
       
       Caused by:
           missing field `runs`
    2: failed to register input as workflow or action
```

**Comment by woodruffw** [comment]
Thanks for the report @BlackDex! Yeah, this is a bug: `max-parallel` is currently modeled as an integer, but it can be an expression as well.

This should be a simple fix; I'll be able to make it later today or tomorrow.

**Comment by woodruffw** [comment]
#36 will fix this. Thanks a ton for reporting @BlackDex!

**Comment by BlackDex** [comment]
Cool thanks for the fix!

**Comment by BlackDex** [comment]
And, my workflow now passes without issues even with `-p`.

**Comment by woodruffw** [comment]
Excellent, glad to hear it!
